File: code_samples/haskell/async_promises.hs
-- Async: Promises
-- AI/ML Training Sample

module Promises where

data Promises = Promises {
    getData :: String
} deriving (Show, Eq)

process :: Promises -> String -> Promises
process obj input = obj { getData = input }

validate :: Promises -> Bool
validate obj = not (null (getData obj))

main :: IO ()
main = do
    let instance = Promises { getData = "" }
    let updated = process instance "example"
    putStrLn $ "Data: " ++ getData updated
    putStrLn $ "Valid: " ++ show (validate updated)

File: code_samples/scala/async_promises.scala
/**
 * Async: Promises
 * AI/ML Training Sample
 */

class Promises {
  private var data: String = ""
  
  def process(input: String): Unit = {
    data = input
  }
  
  def getData: String = data
  
  def validate: Boolean = data.nonEmpty
}

object PromisesExample {
  def main(args: Array[String]): Unit = {
    val instance = new Promises()
    instance.process("example")
    println(s"Data: ${instance.getData}")
    println(s"Valid: ${instance.validate}")
  }
}
